gatling-http-client/src/main/java/io/gatling/http/client/impl/request/WritableRequestBuilder.java

```diff
@@ -41,7 +42,7 @@ private static WritableRequest buildRequestWithoutBody(String url,
                                                          HttpHeaders headers) {
 
     // force content-length to 0 when method usually takes a body, some servers might break otherwise
-    if (!headers.contains(CONTENT_LENGTH) && (method == POST || method == PUT || method == PATCH)) {
+    if (!headers.contains(CONTENT_LENGTH) && asList(POST, PUT, PATCH).contains(method)) {
```

Comment:
Got bitten by this class being in Java and not Scala so == doesn’t mean equals...
Then, your suggested fix allocates a new ArrayList on each execution and then performs an array scan.
Please create a constant Set.